### DISCUSSION TITLE
Supported DNN Models

### DISCUSSION THREAD

**Discussion by ghost** [Motion-Project/motionplus#82] [General]
Hey
Does anyone have any info on what DNN models are known to work ?
I tried using Faster R-CNN and im getting errors.

**Comment by Mr-Dave** [comment]
It has been a while but when I was implementing it, I was using a densenet caffe model.

**Comment by gabbymouth** [comment]
I had some decent results using the MobileNetSSD caffe model. I also did some testing with yolov6s.pt. I still need to spent a lot more time testing different models.

**Comment by ghost** [comment]
Thanks guys,
I tried a bunch of different models to no avail, even with MobileNetSSD (from here https://github.com/amolikvivian/Caffe-SSD-Object-Detection/tree/master/Object%20Detection%20Caffe/Caffe) im getting:
[ERR][ALL][04:cv04:Backyard] algsec_load_dnn: Failed loading model ./SSD_MobileNet.caffemodel
[libprotobuf ERROR google/protobuf/text_format.cc:309] Error parsing text-format opencv_caffe.NetParameter: 1:1: Expected identifier, got: {
[ERR][ALL][03:cv03:Basement] algsec_load_dnn: Error OpenCV(4.5.1) ../modules/dnn/src/caffe/caffe_io.cpp:1157: error: (-2:Unspecified error) FAILED: ReadProtoFromTextFile(param_file, param). Failed to parse NetParameter file: SSD_MobileNet_prototxt.txt in function 'ReadNetParamsFromTextFileOrDie'

[ERR][ALL][03:cv03:Basement] algsec_load_dnn: Failed loading model ./SSD_MobileNet.caffemodel

I tried yolov3, yolov4 as well.

Any ideas on what im doing wrong ? im using MotionPlus version 0.1.1 from .deb release

**Comment by Mr-Dave** [comment]
Maybe use the full path for the model?  Maybe post the Motionplus log section that show the parsed parameters so we can see all the options?

**Comment by ghost** [comment]
Please disregard, i got the SSD_MobileNet model working

**Comment by Mr-Dave** [comment]
Post any tips you have for future users....

**Comment by ghost** [comment]
For the last model SSD_MobileNet.caffemodel it was a silly mistake of not downloading the file properly from github
However im not sure what the issue was with the previous models, i will have to keep experimenting

**Comment by itexpertshire** [comment]
I couldn't make it work with tensorflow model.
1. Downloaded .pb file from http://download.tensorflow.org/models/object_detection/tf2/20200711/ssd_mobilenet_v1_fpn_640x640_coco17_tpu-8.tar.gz and .pbtxt file from https://github.com/tensorflow/models/raw/master/research/object_detection/data/mscoco_label_map.pbtxt.
2. Updated the config file as below
secondary_method dnn
secondary_params model_file=/home/root/pytorch/motioneyeplus/motionplus/model/ssd_mobilenet.pb

After restarting, motionplus displays below error in the log file. 
Jul 25 19:18:13 [ERR][ALL][00:cv00:camera2] algsec_load_dnn: Error OpenCV(4.5.1) ../modules/dnn/src/tensorflow/tf_io.cpp:42: error: (-2:Unspecified error) FAILED: ReadProtoFromBinaryFile(param_file, param). Failed to parse GraphDef file: /home/root/pytorch/motioneyeplus/motionplus/model/ssd_mobilenet.pb in function 'ReadTFNetParamsFromBinaryFileOrDie'

Jul 25 19:18:13 [ERR][ALL][00:cv00:camera2] algsec_load_dnn: Failed loading model /home/root/pytorch/motioneyeplus/motionplus/model/ssd_mobilenet.pb

Appreciate if anyone can suggest direction to resolve this error.

**Comment by gabbymouth** [comment]
This is what I've used with decent results: 

secondary_params model_file=/home/bob/mp/MobileNetSSD_deploy.caffemodel,image_type=full,threshold=0.03,frame_interval=3,config=/home/bob/mp/MobileNetSSD_deploy.prototxt.txt,classes_file=/home/bob/mp/Mo-voc.txt,width=300,height=300,scalefactor=0.007843,mean=127.5

**Comment by gabbymouth** [comment]
BTW, if you are looking for another method, you might want to look into using yolov7's detect.py script and have on_movie_end pass detect.py the folder with picture_output files in it. Also, have on_event_start remove files that were previously saved in it. If you use the yolov7-tiny.pt model it processes images pretty quickly (about 300ms per-image). If you need near real-time results, you can write your own smallish python script to keep the model loaded in memory and process picture_output files as they are created. This is what I am doing now.
